advanced/Scripts/piholeCheckout.sh

```diff
@@ -24,6 +24,93 @@ update="false"
 coltable="/opt/pihole/COL_TABLE"
 source ${coltable}
 
+check_download_exists() {
+  status=$(curl --head --silent "https://ftl.pi-hole.net/${1}" | head -n 1)
+  if echo "$status" | grep -q "404"; then
+    return 1
+  else
+    return 0
+  fi
+}
+
+FTLinstall() {
+  # Download and install FTL binary
+  local binary="${1}"
+  local path="${2}"
+  local str="Installing FTL"
+  echo -ne "  ${INFO} ${str}..."
+
+  if curl -sSL --fail "https://ftl.pi-hole.net/${path}" -o "/tmp/${binary}"; then
+    # Get sha1 of the binary we just downloaded for verification.
+    curl -sSL --fail "https://ftl.pi-hole.net/${path}.sha1" -o "/tmp/${binary}.sha1"
+    # Check if we just downloaded text, or a binary file.
+    cd /tmp
+    if sha1sum --status --quiet -c "${binary}".sha1; then
+      echo -n "transferred... "
+      stop_service pihole-FTL &> /dev/null
+      install -T -m 0755 "/tmp/${binary}" "/usr/bin/pihole-FTL"
+      rm "/tmp/${binary}" "/tmp/${binary}.sha1"
+      start_service pihole-FTL &> /dev/null
+      echo -e "${OVER}  ${TICK} ${str}"
+      return 0
+    else
+      echo -e "${OVER}  ${CROSS} ${str}"
+      echo -e "  ${COL_LIGHT_RED}Error: Download of binary from ftl.pi-hole.net failed${COL_NC}"
+      return 1
+    fi
+  else
+    echo -e "${OVER}  ${CROSS} ${str}"
+    echo -e "  ${COL_LIGHT_RED}Error: URL not found${COL_NC}"
+  fi
+}
+
+get_binary_name() {
+  local machine
+
+  machine=$(uname -m)
+
+  local str="Detecting architecture"
+  echo -ne "  ${INFO} ${str}..."
+  if [[ ${machine} == arm* || ${machine} == *aarch* ]]; then
+    # ARM
+    local rev=$(uname -m | sed "s/[^0-9]//g;")
+    local lib=$(ldd /bin/ls | grep -E '^\s*/lib' | awk '{ print $1 }')
+    if [[ "$lib" == "/lib/ld-linux-aarch64.so.1" ]]; then
```

Comment:
Question: should we not be quoting and bracing strings to be consistent with the rest of the codebase? (e.g: `[[ "${machine}" == "arm"* ||` (There's a couple of instances of differently formatted conditionals, but I won't flag each line for comment in case there's a reason for this!)